src/Polyfill/Polyfill_Task.cs

```diff
@@ -53,23 +54,64 @@ public static Task WaitAsync(
             return Task.FromException(new TimeoutException());
         }
 
-        var delayTask = Task.Delay(timeout, cancellationToken);
+        return WaitAsyncCore(target, timeout, cancellationToken);
+    }
+
+    static async Task WaitAsyncCore(Task task, TimeSpan timeout, CancellationToken cancelToken)
```

Comment:
@stephentoub if you can advise on this implementation please?
This is a polyfill for Task.WaitAsync on TFMs where it doesn't exist.